statsmodels/stats/descriptivestats.py

```diff
@@ -310,9 +312,83 @@ def sign_test(self, samp, mu0=0):
     #def sign_test_p(samp,mu0=0):
         #return self.sign_test(samp,mu0)[1]
 
+class DescrStats(object):
+    '''descriptive statistics for data.
+
+    Parameters
+    ----------
+    dataset : ndarray or DataFrame
+        dataset for descriptive statistics.
+    '''
+    def __init__(self, data):
+        '''convert data to pandas dataframe and extract the
+         numeric values
+        '''
+        if isinstance(data, pd.DataFrame) is False:
+            self.data = pd.DataFrame(data)
+        else:
+            self.data = data
+        # select only numerics
+        numerics = ['int16', 'int32', 'int64', 'float16', 'float32', 'float64']
+        self.data = self.data.select_dtypes(include=numerics)
+
+    @OneTimeProperty
+    def obs(self):
+        '''return number of observations'''
+        return self.data.shape[0]
+
+    @OneTimeProperty
+    def mean(self):
+        '''mean of the data'''
+        return np.mean(self.data).values
+
+    @OneTimeProperty
+    def var(self):
+        '''variance of the data'''
+        return np.var(self.data, axis=0).values
+
+    @OneTimeProperty
+    def std(self):
+        '''standard deviation of the data'''
+        return np.std(self.data, axis=0).values
+
+
+    def summary_frame(self, stats='basic'):
+        '''collect all the stats and return as a DataFrame
+
+        Parameters
+        -----------
+        stats: str
+            The desired statistics, Accepts 'basic' or 'all' or a list.
+               'basic' = ('obs', 'mean', 'std', 'var')
+
+        Returns
+        -------
+        df : pandas dataframe
+        '''
+        #TODO: Add support for all and list of stats
+        if stats == 'basic':
+            stats = ['obs', 'mean', 'std', 'var']
+            #TODO: Hack around to make this simpler
+            obs = self.obs
+            mean = self.mean
+            std = self.std
+            var = self.var
+
+        return pd.DataFrame(data=[[obs, mean, std, var]],
```

Comment:
if the individual methods/ properties return pandas Series, then this would need to be a concatenation of Series